csharp/src/Ice/IceClassFactory.cs

```diff
@@ -0,0 +1,10 @@
+//
+// Copyright (c) ZeroC, Inc. All rights reserved.
+//
+
+namespace ZeroC.Ice
+{
+    /// <summary>Delegate use by InputStream to create class instances</summary>
+    /// <returns>The new class instance.</returns>
+    internal delegate AnyClass IceClassFactory();
```

Comment:
Why is the Ice prefixed required here?

Reply:
We could define this type Communicator.cs instead of having a file just for this.

Reply:
Ice prefix is required because we generate factories in `ZeroC.Ice.ClassFactory` namespace

Reply:
I missed that in my review as well. Better switch to a simple Func<>.